src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1541,31 +1544,55 @@
   ;; we've encountered during the migration to avoid inserting
   ;; duplicates into the new table.
 
-  (let [event-count (-> "select count(*) from resource_events"
-                        jdbc/query-to-vec first :count)
-        last-logged (atom (.getTime (java.util.Date.)))]
-    (jdbc/call-with-query-rows
-     ["select
-       report_id, certname_id, status, \"timestamp\", resource_type, resource_title, property,
-       new_value, old_value, message, file, line, containment_path, containing_class, corrective_change
-     from resource_events"]
-     (fn [rows]
-       (reduce (fn [hashes-seen [row i]]
-                 (let [now (.getTime (java.util.Date.))]
-                   (when (> (- now @last-logged) 60000)
-                     (maplog :info
-                             {:migration 69 :at i :of event-count}
-                             #(trs "Migrated {0} of {1} events" (:at %) (:of %)))
-                     (reset! last-logged now)))
-                 (conj hashes-seen
-                       (let [hash-str (hash/resource-event-identity-pkey row)]
-                         (when-not (hashes-seen hash-str)
-                           (jdbc/insert-multi! "resource_events_transform"
-                                               (list (assoc row :event_hash
-                                                            (sutils/munge-hash-for-storage hash-str)))))
-                         hash-str)))
+  (jdbc/call-with-query-rows
+   ["select
+     report_id, certname_id, status, \"timestamp\", resource_type, resource_title, property,
+     new_value, old_value, message, file, line, containment_path, containing_class, corrective_change
+   from resource_events"]
+   (fn [rows]
+     (let [event-count (-> "select count(*) from resource_events"
```

Comment:
This is going to run the `select count(*)` for every batch. This should be outside of the `call-with-query-rows` like it was originally.

Reply:
Hmm, doesn't `call-with-query-rows` invoke the `(fn [rows] ...)` once and pass a sequence of rows returned by the query?

The batching bit happens inside the `let` where `rows` is passed to `reduce`.

Reply:
Yeah, unless I'm misreading, I think this should be fine since `rows` is the entire `seq` of rows, and there's only one of those -- now if it were inside the inter reduce...